realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -531,11 +532,83 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
                         field.equals(NullTypes.FIELD_DATE_NULL)) {
                     assertEquals("Field '" + field + "' is required. Either set @Required to field '" +
                             field + "' " +
-                            "or migrate using io.realm.internal.Table.convertColumnToNullable().", e.getMessage());
+                            "or migrate using RealmObjectSchema.setNullable().", e.getMessage());
                 } else {
                     assertEquals("Field '" + field + "' does not support null values in the existing Realm file."
                                     + " Either set @Required, use the primitive type for field '"
-                                    + field + "' or migrate using io.realm.internal.Table.convertColumnToNullable().",  e.getMessage());
+                                    + field + "' or migrate using RealmObjectSchema.setNullable().",  e.getMessage());
+                }
+            }
+        }
+    }
+
+    // Setting the existing @PrimaryKey field nullable to migrate
```

Comment:
Not sure I understand this? This test is testing that you need to set fields as nullable, right?
Maybe rephrase to "Test that older Realms with Boxed type primary keys needs to be migrated to be nullable in order to support support Realm 0.89+" or something like that?